src/components/PopupConfirmDeletion.js

```diff
@@ -0,0 +1,36 @@
+import Popup from "./Popup";
+export default class PopupConfirmDeletion extends Popup {
+  constructor({ popupSelector, handleFormSubmit }) {
+    super({ popupSelector });
+    this._handleFormSubmit = handleFormSubmit;
+    this._deleteButton = this._popupElement.querySelector(
+      ".card__delete-button"
+    );
+    this._isSubmitted = false;
+    this._cardId = null;
+    this._card = null;
+  }
+
+  open(card, cardId) {
+    this._card = card;
+    this._cardId = cardId;
+    super.open();
+  }
+
+  getIsSubmitted() {
+    return this._isSubmitted;
+  }
+
+  confirmDeletion(api) {
+    this._handleFormSubmit = api;
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._popupElement.addEventListener("click", (evt) => {
```

Comment:
* When I close the `confirmation` modal, the card gets **deleted**, but only after clicking on the `Yes` button it should be deleted


Please, add this click to `this._deleteButton ` rather than the whole popup


```js

    this._deleteButton.addEventListener("click", (evt) => {

```


But you need to correctly find `this._deleteButton` in line 6 because it's `null` now

Reply:
Knew I missed something. Should work now. Fantastic GIF for the profile picture by the way - had me dying of laughter.  I might keep it that way.